Source/Immutable/Private/Immutable/IOS/ImmutableIOS.cpp

```diff
@@ -0,0 +1,52 @@
+#include "ImmutableIOS.h"
+#include "Immutable/ImmutablePassport.h"
+
+API_AVAILABLE(ios(12.0))
+ASWebAuthenticationSession *_authSession;
+
+@implementation ImmutableIOS
+
+- (instancetype)init
+{
+	self = [super init];
+	return self;
+}
+
++ (ImmutableIOS *)instance
+{
+	static ImmutableIOS *staticImmutableIOS;
+	static dispatch_once_t once;
+	dispatch_once(&once, ^{
+		staticImmutableIOS = [[self alloc] init];
+	});
+	return staticImmutableIOS;
+}
+
+
+- (void) launchUrl:(const char*)url
+{
+	NSURL * URL = [NSURL URLWithString: [[NSString alloc] initWithUTF8String:url]];
+	NSString *bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+
+	ASWebAuthenticationSession* authSession = [[ASWebAuthenticationSession alloc] initWithURL:URL callbackURLScheme:bundleIdentifier completionHandler:^(NSURL * _Nullable callbackURL, NSError * _Nullable error) {
+        
```

Comment:
maybe we reset the session flag since it's not needed anymore? It's my understanding it's a global variable, so it shouldnt have any retain cycles?

```suggestion
        _authSession = nil;
        
```